### ISSUE TITLE
Support direct wifi Connector devices 

### ISSUE THREAD

**Issue by elvenwhite** [gormanb/homebridge-connector-hub#4]
**Is your feature request related to a problem? Please describe:**

Hello, 

My Connector app has three devices atm : one blind(RF connected to Connector hub) and 2 curtains(DT-72TE/S wifi motor). With your plugin, I was able to add the blind, but curtains are not being exposed to Home app because they aren't connected to the hub.

**Describe the solution you'd like:**

AFIAK, DT-72TE/S motors are connected with similar protocols with the hub, therefore I think it can be connected into HB with same manner. Do you have any plan to expand the plugin to support those wifi motors?

**Describe alternatives you've considered:**

**Additional context:**

**Comment by gormanb** [comment]
Hi @elvenwhite!

I don't have any WiFi motor devices, so I'm not able to investigate this myself. If you're willing to test out some beta versions on your system and give me feedback on what works and what doesn't, I'd be happy to add support for multiple devices - assuming that they're using the same protocol as the hub itself.

Does that sound OK to you?

**Comment by elvenwhite** [comment]
That would be grateful. I tried to put ip address of WiFi motor for hub ip, and debug message shows they share same protocol. I think just adding capability of supporting multiple hubs might solve the problem. 

`
//single wifi motor
[2/6/2023, 7:54:41 AM] [ConnectorHub] [DEBUG] Discovered devices: {
  msgType: 'GetDeviceListAck',
  mac: 'c45bbe4483a5',
  deviceType: '22000000',
  fwVersion: '0.8.4.0',
  ProtocolVersion: '0.9',
  token: '37412C478E0FBEAB',
  data: [ { mac: 'c45bbe4483a5', deviceType: '22000000' } ]
}

// hub
[2/6/2023, 8:04:55 AM] [ConnectorHub] [DEBUG] Discovered devices: {
  msgType: 'GetDeviceListAck',
  mac: 'a4e57cb06344',
  deviceType: '02000001',
  fwVersion: 'A1.0.9_B0.2.0',
  ProtocolVersion: '0.9',
  token: '967DC7CD55376DA9',
  data: [
    { mac: 'a4e57cb06344', deviceType: '02000001' },
    { mac: 'a4e57cb063440001', deviceType: '10000000' }
  ]
}`

**Comment by gormanb** [comment]
Hi @elvenwhite - I just published `1.0.7-beta.0`, which should be available now in Homebridge - and Hoobs, if you're using that.

If you're using Homebridge:
- Go to the `Plugins` tab on the UI, click the wrench icon in the bottom-right of the `Homebridge Connector Hub` plugin, select `Install Alternate Version`, install `1.0.7-beta.0` from the drop-down list, and restart Homebridge.
- After the restart, you should be able to open the plugin's `Settings` screen and there will be a new section called `Connector Hub / Wifi Device IPs`. Add the IPs of each of your devices in there, one in each box - click the `Add To Connector Hub / Wifi Device IPs` button to create new entries.

If you're using Hoobs:
- Go to the plugin page for `Connector Hub` on whichever bridge you installed it into, select the `Versions` tab, click `1.0.7-beta.0`, and install it into a new bridge or the existing bridge. If you install it into a new bridge, disable the old bridge while you test the beta version.
- Unfortunately, it seems that the new configuration screen isn't available in Hoobs when you install the beta, so you'll have to go to the `Advanced Config` and make sure it looks like this:
```
{
    "hubIps": [
        "your-device-ip-1",
        "your-device-ip-2",
        "your-device-ip-3"
    ],
    "connectorKey": "your-connector-key",
    "enableDebugLog": false,
    "platform": "ConnectorHub"
}
```

Please give that a try, and let me know if you're able to use all of your devices in Homekit. If not, please let me know exactly what happened, and provide logs from your test-runs.

**Comment by elvenwhite** [comment]
Thanks for update! I tried with Homebridge, and got following result : 

- Devices connected by connector hub works as before.
- Devices using wifi are displayed at accessories tab in homebridge as following.

![image](https://user-images.githubusercontent.com/532910/218307953-af64c447-96d0-427a-8ab7-75f6baf2f05e.png)

- Devices using wifi are not showing up in homekit.  They can be activated by homebridge UI, but status is not displayed correctly(only sometimes give correct status). Log file is attached.

[homebridge.log.txt](https://github.com/gormanb/homebridge-connector-hub/files/10715928/homebridge.log.txt)

**Comment by gormanb** [comment]
Hi @elvenwhite - thanks for the additional information!

The reason they're showing up as "Unidentified Device" is because it looks like the Wifi curtains don't expose their model type in the same way as devices connected to a hub. I can fix that easily, but it shouldn't actually affect the behaviour of the devices - only the display name in Homebridge and Homekit.

I don't have any explanation for why the accessories aren't showing up in Homekit. The logs indicate that they're being correctly detected and registered, and as you mentioned they're showing up in the Homebridge UI:
```
[2/12/2023, 8:07:42 PM] [ConnectorHub] Adding new accessory: Connector Device c45bbe453878
[2/12/2023, 8:07:42 PM] [ConnectorHub] Adding new accessory: Connector Device c45bbe4483a5
```
After the accessory is registered, it's up to Homebridge to make it visible to Homekit. There don't appear to be any errors or warnings in the logs to indicate a problem with registering the accessory, so the only thing I can think of is that there's some latency in Homekit detecting the updated list of accessories attached to the Homebridge. Are you using the Home app? Have you checked to see if the new accessories appear in third-party apps like the Eve app?

You also mentioned that the curtains "can be activated by homebridge UI, but status is not displayed correctly(only sometimes give correct status)." Can you provide more details about this? What happens when you activate the devices? What incorrect statuses are being displayed?

**Comment by gormanb** [comment]
FYI, I've just published `v1.0.7-beta.1` which should resolve the "Unidentified Device" issue. If you can let me know how the devices are behaving when you use them via the Homebridge UI (i.e. what incorrect statuses they're showing) then I should hopefully be able to fix that as well.

**Comment by elvenwhite** [comment]
beta.1 works as resolving `Unidentified Devices`. Also, your guess about latency in Homekit detection was right! I could find new devices from Home app. 

The status problem is, the devices status are initially `opening` after being added. I close them with Home app, then their status turns `closing->closed` while they are being closed, and `opened` after 2, 3 seconds from fully closed. Logs are:

`[2/13/2023, 12:16:38 PM] [ConnectorHub] Targeted: [ 'Wi-Fi Curtain 01:c45bbe4483a5', 0 ]
[2/13/2023, 12:16:38 PM] [ConnectorHub] [DEBUG] Target response: {
  msgType: 'WriteDeviceAck',
  mac: 'c45bbe4483a5',
  deviceType: '22000000',
  msgID: '20230213031638782',
  data: {
    operation: 2,
    direction: 0,
    currentPosition: 100,
    targetPosition: 100,
    currentState: 3,
    switchMode: 0,
    controlMode: 0,
    RSSI: -40
  }
}
[2/13/2023, 12:16:50 PM] [ConnectorHub] [DEBUG] Updated Wi-Fi Curtain 01:c45bbe4483a5 state: {
  msgType: 'ReadDeviceAck',
  mac: 'c45bbe4483a5',
  deviceType: '22000000',
  msgID: '20230213031648384',
  data: {
    operation: 2,
    direction: 0,
    currentPosition: 0,
    targetPosition: 0,
    currentState: 3,
    switchMode: 0,
    controlMode: 0,
    RSSI: -40
  }
}
[2/13/2023, 12:16:50 PM] [ConnectorHub] Updating position: [ 'Wi-Fi Curtain 01:c45bbe4483a5', 100 ]`

**Comment by gormanb** [comment]
Hi @elvenwhite,

Having examined the full logs you posted earlier, I believe I know why the status on your devices are not showing up correctly. For hub devices - at least for blinds - the position and target values are the inverse of Homekit; the hub considers 0 to be fully open and 100 to be closed, while Homekit treats 0 as closed and 100 as fully open. The plugin therefore translates the values between the Homekit and hub versions to keep both systems working smoothly together.

However, it looks like for your WiFi curtains, the position and target values are **_the same_** as Homekit; that is, 0 is closed and 100 is open. Therefore, when the hub tries to open or close the curtains and sees the device report the _opposite_ of what it expects, it reports the wrong status back to Homekit.

To investigate this theory, I've published `1.0.7-beta.2`. Please install that, restart Homebridge, and try your curtains again to see if the statuses are now reported correctly. If not, please let me know what you see!

**Comment by gormanb** [comment]
Hi @elvenwhite - I've just published `1.0.7-beta.3`, which I believe should fix the targeting direction. Please give it a try and let me know whether your curtain behaves as expected!
